scripts/globals/abilities/reverse_flourish.lua

```diff
@@ -12,47 +12,34 @@ require("scripts/globals/msg")
 
 function onAbilityCheck(player,target,ability)
 
-    if (player:hasStatusEffect(tpz.effect.FINISHING_MOVE_1)) then
+    if
+        player:hasStatusEffect(tpz.effect.FINISHING_MOVE_1) or
+        player:hasStatusEffect(tpz.effect.FINISHING_MOVE_2) or
+        player:hasStatusEffect(tpz.effect.FINISHING_MOVE_3) or
+        player:hasStatusEffect(tpz.effect.FINISHING_MOVE_4) or
+        player:hasStatusEffect(tpz.effect.FINISHING_MOVE_5)
+    then
         return 0,0
-
-    elseif (player:hasStatusEffect(tpz.effect.FINISHING_MOVE_2)) then
-        return 0,0
-
-    elseif (player:hasStatusEffect(tpz.effect.FINISHING_MOVE_3)) then
-        return 0,0
-
-    elseif (player:hasStatusEffect(tpz.effect.FINISHING_MOVE_4)) then
-        return 0,0
-
-    elseif (player:hasStatusEffect(tpz.effect.FINISHING_MOVE_5)) then
-        return 0,0
-
```

Comment:
This logic comparison forces 5 checks every time. The old only makes checks until it finds a true statement, somewhere between 1 and 5 checks.